cmd/notation/list.go

```diff
@@ -33,8 +32,8 @@ func listCommand(opts *listOpts) *cobra.Command {
 		Short:   "List signatures of the signed artifact",
 		Long:    "List all the signatures associated with signed artifact",
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) == 0 {
-				return errors.New("no reference specified")
+			if err := arg.ValidateCount(args, 1, "missing reference"); err != nil {
+				return err
```

Comment:
The cobra package has the following method: [ExactArgs](https://pkg.go.dev/github.com/spf13/cobra#ExactArgs). Therefore, we can have something like this:
```
Args: cobra.ExactArgs(1),
PreRunE: func(cmd *cobra.Command, args []string) error {
			if opts.ociLayout {
				opts.inputType = inputTypeOCILayout
			}
            opts.reference = args[0]
			return experimental.CheckFlagsAndWarn(cmd, "oci-layout")
		},
```
Similar applies to the other commands.

Reply:
I did try using [ositional and Custom Arguments](https://github.com/spf13/cobra/blob/main/user_guide.md#positional-and-custom-arguments) but the error messages were not user friendly. Also, we need to rewrite bunch of integ tests to support this.

```
./notation verify
Error: accepts 1 arg(s), received 0

./notation verify someImage -m "adasd=1" sadad=2
Error: accepts 1 arg(s), received 2
```

Reply:
> I did try using [positional and Custom Arguments](https://github.com/spf13/cobra/blob/main/user_guide.md#positional-and-custom-arguments) but the error messages were not user friendly. Also, we would need to rewrite bunch of unit tests to support this because they verifies that passed arguments are stored in `opts`.
> 
> ```
> ./notation verify
> Error: accepts 1 arg(s), received 0
> 
> ./notation verify someImage -m "adasd=1" sadad=2
> Error: accepts 1 arg(s), received 2
> ```

I see, you are trying to customize the error messages here. Makes sense to me.